### PR TITLE
Draft pull request template

### PR DESCRIPTION
**Pull request by kevingranade**
Adds a rough draft of a pull request template.
Main goal is outlining the new summary line.

Related to #21325 

SUMMARY: Infrastructure "Added pull request template."

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+#### Summary
+A one-line description of your change that will be extracted and added to the [project changelog](../data/changelog.txt).  
+The format is ```SUMMARY: Category "description"```  
+Category is one of Features, Content, Interface, Mods, Balance, Bugfixes, Performance, Infrastructure, Build, I18N
+
+#### Purpose of change
+If theres an existing issue describing the problem this PR addresses or the feature it adds, please link it like #1234  
+If it *fully* resolves an issue, link it like ```Fixes #1234```  
+Even if the issue describes the problem, please prove a few-sentence summary here.
```
Comment (alanbrady) [inline .github/pull_request_template.md]: ~~prove~~ provide?

Comment (kevingranade) [reply .github/pull_request_template.md]: Nope, prove it!

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+#### Summary
+A one-line description of your change that will be extracted and added to the [project changelog](../data/changelog.txt).  
+The format is ```SUMMARY: Category "description"```  
+Category is one of Features, Content, Interface, Mods, Balance, Bugfixes, Performance, Infrastructure, Build, I18N
+
+#### Purpose of change
+If theres an existing issue describing the problem this PR addresses or the feature it adds, please link it like #1234  
+If it *fully* resolves an issue, link it like ```Fixes #1234```  
+Even if the issue describes the problem, please provide a few-sentence summary here.  
+If there is no related issue, please describe the issue you are addressing,  including how to trigger a bug if this is a bugfix.
```
Comment (alanbrady) [inline .github/pull_request_template.md]: nitpick, there's two spaces after 'addressing,'

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+### how to use
```
Comment (Night-Pryanik) [inline .github/pull_request_template.md]: Capitalize `how`.

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+Leave the headings unless they dont apply to your PR, replace the explanatory text with text describing your PR.
```
Comment (Night-Pryanik) [inline .github/pull_request_template.md]: `dont` -> `don't`.

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+#### Summary
+A one-line description of your change that will be extracted and added to the [project changelog](../data/changelog.txt).  
+The format is ```SUMMARY: Category "description"```  
+Category is one of Features, Content, Interface, Mods, Balance, Bugfixes, Performance, Infrastructure, Build, I18N
+
+#### Purpose of change
+If theres an existing issue describing the problem this PR addresses or the feature it adds, please link it like #1234
```
Comment (Night-Pryanik) [inline .github/pull_request_template.md]: `theres` -> `there's`.

File: .github/pull_request_template.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
+### how to use
+Leave the headings unless they dont apply to your PR, replace the explanatory text with text describing your PR.
+
+#### Summary
+A one-line description of your change that will be extracted and added to the [project changelog](../data/changelog.txt).  
+The format is ```SUMMARY: Category "description"```  
+Category is one of Features, Content, Interface, Mods, Balance, Bugfixes, Performance, Infrastructure, Build, I18N
```
Comment (Night-Pryanik) [inline .github/pull_request_template.md]: Is this case-sensitive?

**Comment by alanbrady** [comment]
Would it make sense to add the available "Categories" a SUMMARY can be in, here as well?

File: .github/pull_request_template.md (hunk 6 above)
Comment (kevingranade) [reply .github/pull_request_template.md]: It is currently, but it doesn't have to be.

**Comment by kevingranade** [comment]
I dont follow, the categories are listed.

File: .github/pull_request_template.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+#### Purpose of change
+If theres an existing issue describing the problem this PR addresses or the feature it adds, please link it like #1234  
+If it *fully* resolves an issue, link it like ```Fixes #1234```  
+Even if the issue describes the problem, please provide a few-sentence summary here.  
+If there is no related issue, please describe the issue you are addressing, including how to trigger a bug if this is a bugfix.
+
+#### Describe the solution
+How does the feature work, or how does this fix a bug?  
+The easier you make your solution to understand, the faster it can get merged.
```
Comment (DracoGriffin) [inline .github/pull_request_template.md]: Suggest: `The easier you make your solution to understand` to `The easier your solution is to understand`

File: .github/pull_request_template.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
+### How to use
+Leave the headings unless they don't apply to your PR, replace the explanatory text with text describing your PR.
+
+#### Summary
+A one-line description of your change that will be extracted and added to the [project changelog](../data/changelog.txt).
```
[the remark this answers is not in the document]
Comment (kevingranade) [reply .github/pull_request_template.md]: Two trailing spaces is github markdown for newline.

File: .github/pull_request_template.md [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+
+#### Summary
+A one-line description of your change that will be extracted and added to the [project changelog](../data/changelog.txt).  
+The format is ```SUMMARY: Category "description"```  
+Category is one of Features, Content, Interface, Mods, Balance, Bugfixes, Performance, Infrastructure, Build, I18N
+
+#### Purpose of change
+If theres an existing issue describing the problem this PR addresses or the feature it adds, please link it like #1234  
+If it *fully* resolves an issue, link it like ```Fixes #1234```  
+Even if the issue describes the problem, please provide a few-sentence summary here.
```
[the remark this answers is not in the document]
Comment (kevingranade) [reply .github/pull_request_template.md]: Disagree, few-sentence gives a scale, short could range from a single word to a 10 sentence paragraph.

File: .github/pull_request_template.md (hunk 7 above)
Comment (kevingranade) [reply .github/pull_request_template.md]: I prefer the active voice, even if their solution is difficult, this is their opportunity to *make* it easier to understand.

File: .github/pull_request_template.md [hunk 10]
```diff
@@ -0,0 +1,25 @@
+### How to use
+Leave the headings unless they don't apply to your PR; replace the explanatory text with text describing your PR.
+
+#### Summary
+A one-line description of your change that will be extracted and added to the [project changelog](../data/changelog.txt).  
+The format is: ```SUMMARY: Category "description"```  
+The categories to choose from are: Features, Content, Interface, Mods, Balance, Bugfixes, Performance, Infrastructure, Build, I18N
```
Comment (mlangsdorf) [inline .github/pull_request_template.md]: Is there a description of these categories somewhere? I'm not sure of the difference between Features and Content or between Infrastructure and Build. I get that it's going to be a judgment call, but more explanation is necessary.

There should also be a category for Documentation only fixes, unless those are supposed to go under Interface or Infrastructure.

Two particular issues come to mind:
1. Code reorganization/refactoring. Build or Infrastructure?
2. Code that adds support to allow modders to do something, and gives an example. Is that Features, Content, Mods, or Infrastructure?

Comment (ZhilkinSerg) [reply .github/pull_request_template.md]: Will there be correlation between Categories and current set of Labels?

Comment (kevingranade) [reply .github/pull_request_template.md]: I think if we document each category, it's going to need to go in it's own document, that's going to be way too verbose for the template, just from the point of view of people having to delete all that content before entering their own.
I'll sketch something out at docs/ChangelogGuidelines.md

Comment (mlangsdorf) [reply .github/pull_request_template.md]: The changelog guidelines look good to me. They're not too long but they're still comprehensive. Some stuff is still going to be a judgment call, but it's a judgment call that people have enough information to make.

**Pull request merged**